apps/main/src/loan/components/PageCrvUsdStaking/index.tsx

```diff
@@ -7,41 +6,64 @@ import useStore from '@/loan/store/useStore'
 import StatsBanner from '@/loan/components/PageCrvUsdStaking/StatsBanner'
 import DepositWithdraw from '@/loan/components/PageCrvUsdStaking/DepositWithdraw'
 import UserInformation from '@/loan/components/PageCrvUsdStaking/UserInformation'
-import UserPositionBanner from '@/loan/components/PageCrvUsdStaking/UserPositionBanner'
+import UserPosition from '@/loan/components/PageCrvUsdStaking/UserPosition'
+import Statistics from '@/loan/components/PageCrvUsdStaking/Statistics'
 import { useWallet } from '@ui-kit/features/connect-wallet'
-
-const CrvUsdStaking = ({ mobileBreakpoint }: { mobileBreakpoint: string }) => {
-  const {
-    fetchUserBalances,
-    checkApproval,
-    inputAmount,
-    fetchExchangeRate,
-    fetchCrvUsdSupplies,
-    fetchSavingsYield,
-    stakingModule,
-  } = useStore((state) => state.scrvusd)
+import { useSwitch } from '@ui-kit/hooks/useSwitch'
+import { Stack, useMediaQuery } from '@mui/material'
+import { Sizing } from '@ui-kit/themes/design/0_primitives'
+import { SizesAndSpaces } from '@ui-kit/themes/design/1_sizes_spaces'
+import Fade from '@mui/material/Fade'
+import { useScrvUsdUserBalances } from '@/loan/entities/scrvusdUserBalances'
+
+const { MaxWidth } = SizesAndSpaces
+
+const CrvUsdStaking = () => {
+  const [isChartExpanded = false, , minimizeChart, toggleChartExpanded] = useSwitch(false)
+  const checkApproval = useStore((state) => state.scrvusd.checkApproval)
+  const inputAmount = useStore((state) => state.scrvusd.inputAmount)
+  const fetchExchangeRate = useStore((state) => state.scrvusd.fetchExchangeRate)
+  const fetchCrvUsdSupplies = useStore((state) => state.scrvusd.fetchCrvUsdSupplies)
+  const stakingModule = useStore((state) => state.scrvusd.stakingModule)
   const lendApi = useStore((state) => state.lendApi)
-  const { signerAddress, provider } = useWallet()
+  const { signerAddress, connecting, walletName } = useWallet()
   const chainId = useStore((state) => state.curve?.chainId)
-  const userScrvUsdBalance = useStore((state) => state.scrvusd.userBalances[signerAddress ?? '']?.scrvUSD) ?? '0'
-  const isUserScrvUsdBalanceZero = BigNumber(userScrvUsdBalance).isZero()
+
+  const {
+    data: userScrvUsdBalance,
+    isFetching: isUserScrvUsdBalanceFetching,
+    isFetched: isUserScrvUsdBalanceFetched,
+    refetch: refetchUserScrvUsdBalance,
+  } = useScrvUsdUserBalances({
+    signerAddress: signerAddress ?? '',
+  })
+
+  const isUserScrvUsdBalanceZero =
+    signerAddress && userScrvUsdBalance ? BigNumber(userScrvUsdBalance.scrvUSD).isZero() : true
+
+  const connectedUserNoScrvUsdBalance =
+    !!signerAddress && !!walletName && !!isUserScrvUsdBalanceFetched && !!isUserScrvUsdBalanceZero
+
+  // walletName indicates the wallet is cached and will begin connecting
+  const showStatsBanner =
+    !walletName || connectedUserNoScrvUsdBalance || !signerAddress
+      ? true
+      : !connecting && !isUserScrvUsdBalanceFetching && !walletName && isUserScrvUsdBalanceZero
```

Comment:
Note `walletName` cannot be false in the 2nd part
```suggestion
    !walletName || connectedUserNoScrvUsdBalance || !signerAddress
      || (!connecting && !isUserScrvUsdBalanceFetching && isUserScrvUsdBalanceZero)
```

Reply:
this check was a head scratcher 😫 thanks, I will remove